### DISCUSSION TITLE
How do I know If workflow is triggered by git branch or git tag?

### DISCUSSION THREAD

**Discussion by cytopia** [community/community#26731] [Actions]
<p>My workflow run is defined as follows:</p>
<pre><code>on:
  pull_request:
  push:
    branches: ['**']
    tags: ['**']

  # Runs daily
  schedule:
    - cron: '0 0 * * *'
</code></pre>
<p>How do I know inside my steps that I am currently on a <strong>git tag</strong> or <strong>git branch</strong>? And how do I get the name of either of it?</p>
<p><strong>Goal:</strong></p>
<p>If the workflow is triggered by a git tag, I want to tag my Docker images by the git tag name.<br>
If the workflow is triggered by a git branch, I want to tag my Docker images by the git branch name plus a suffix.</p>

**Comment by eddiechenms** [comment, marked as answer]
<p>You can use <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#contexts" rel="nofollow noopener">Context</a> to access the information about the workflow run which will include the information you want. For information about trigger, you can use <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">github context</a>: github.ref. Usually, if the workflow is triggered by a tag, the value of github.ref would be something like “refs/tags/xxx”, and if the workflow is triggered by a branch, the value would be “refs/heads/xxx”. And then you can choose your next steps base on this information.</p>

**Comment by Daniele-Tentoni** [comment]
<p>Hello eddie,<br>
thanks for your solution. I have a question: how can I run a workflow only for tags on main branch?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="Daniele-Tentoni" data-post="3" data-topic="16751">
<div class="title">
<div class="quote-controls"></div>
 Daniele-Tentoni:</div>
<blockquote>
<p>how can I run a workflow only for tags on main branch?</p>
</blockquote>
</aside>
<p>Git tags are never on a branch, they refer to specific commits independently of branches. Can you clarify what the exact condition is that you’re looking for? Possibly something like “<a href="https://git-scm.com/docs/git-branch#Documentation/git-branch.txt---containsltcommitgt">does the branch contain</a> the commit the tag points at”?</p>

**Comment by Daniele-Tentoni** [comment]
<p>Yes, I was looking for it. My issue was to understand when the commit referred by the tag is in a specific branch. You link had explained me that right. Thank you very much.</p>
